addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2563,6 +2566,41 @@ const Many2oneUserValueWidget = SelectUserValueWidget.extend({
         }
         return this._super(...arguments);
     },
+    /**
+     * Updates the domain with defined inclusive filter to make sure that only
+     * records that are linked to specific records are retrieved.
+     * Filtering-in is configured with
+     *   * a `filterInModel` attribute, the linked model
+     *   * a `filterInField` attribute, field of the linked model holding
+     *   allowed values for this widget
+     *
+     * @param {integer[]} linkedRecordsIds
+     * @returns {Promise}
+     */
+    async setFilterInDomainIds(linkedRecordsIds) {
+        if (this.options.filterInField && this.options.domain.length) {
+            const allowedIds = new Set();
+            if (linkedRecordsIds) {
+                const parentRecordsData = await this._rpc({
+                    model: this.options.filterInModel,
+                    method: 'search_read',
+                    fields: [this.options.filterInField],
+                    domain: [['id', 'in', linkedRecordsIds]],
+                });
+                parentRecordsData.forEach(record => {
+                    record[this.options.filterInField].forEach(item => allowedIds.add(item));
+                });
+            }
+            if (this.allowedIDsInDomain) {
+                this.options.domain.pop();
+                this.allowedIDsInDomain = false;
+            }
+            if (allowedIds.size) {
+                this.options.domain.push(['id', 'in', [...allowedIds]]);
+                this.allowedIDsInDomain = true;
+            }
```

Comment:
Something like that is good idea I think indeed, to test :+1: (but instead of `[...a, ...b]`, just use `a.concat(b)` :wink:)

Also, when you reply back to me to a week old comment, try to ping me explicitly if you want to make sure I see it. For PR outside of my team, I don't always check if the dozens of mail I receive every day are addressed to me or not if I don't see an explicit ping :wink:  (here it's really a miracle I opened the PR to check ^^)